ql/src/java/org/apache/hadoop/hive/ql/io/parquet/ParquetRecordReaderBase.java

```diff
@@ -209,6 +214,22 @@ public FilterCompat.Filter setFilter(final JobConf conf, MessageType schema) {
     }
   }
 
+  private MessageType getSchemaWithoutPartitionColumns(JobConf conf, MessageType schema) {
+    String partCols = conf.get(IOConstants.PARTITION_COLUMNS);
+    if (partCols != null && partCols.length() > 0) {
+      Set<String> partitionColumns = new HashSet<>(Arrays.asList(partCols.split(",")));
+      List<Type> newFields = new ArrayList<>();
+
+      for (Type field: schema.getFields()) {
+        if(!partitionColumns.contains(field.getName())) {
```

Comment:
Could you confirm if this comparison works in a case-insensitive way ? i.e if the partition columns is defined as 'part_col' and the query has predicate on 'PART_COL'.

Reply:
I tested with `PCOL` and  \`pCol\`, and it's working fine. I have added more queries to the q file.